### ISSUE TITLE
A question about  about the validation code

### ISSUE THREAD

**Issue by Italy2006** [qjadud1994/DRS#11]
Thank you for providing such a convenient code, but I have some questions about it. On the line 75 of eval.py, you directly use the image after crop for performance verification. Is this accurate? Another question is why not apply ms+flip to inference?

**Comment by qjadud1994** [comment]
Are you talking about the [DeepLabV3 code](https://github.com/qjadud1994/DRS/blob/866f0e799a4e2a8b7d84016ee604f9bb0ce7f572/DeepLabV3Plus-Pytorch/eval.py#L75)?

As I mentioned, we strictly followed the DeepLab-V3+ pytorch implementation from https://github.com/VainF/DeepLabV3Plus-Pytorch, and trained with our pseudo labels.

**Comment by Italy2006** [comment]
But I looked at his Deeplabv2 library, it does not use resize_eval. I read other articles, none of the libraries use Resize_eval. If you use Resize_eval, the model results will be significantly improved

**Comment by Italy2006** [comment]
> Are you talking about the [DeepLabV3 code](https://github.com/qjadud1994/DRS/blob/866f0e799a4e2a8b7d84016ee604f9bb0ce7f572/DeepLabV3Plus-Pytorch/eval.py#L75)?
> 
> As I mentioned, we strictly followed the DeepLab-V3+ pytorch implementation from https://github.com/VainF/DeepLabV3Plus-Pytorch, and trained with our pseudo labels.

Another question is, if you use resize_eval, what do you do when you submit the test set?

**Comment by qjadud1994** [comment]
For DeepLab-V2, we did not use Resize_eval.

The result of DeepLab-V3 is our additional experiment after publishing, so we skip the test-mIoU and report only the val-mIoU following the [DeepLabV3 implementation])(https://github.com/VainF/DeepLabV3Plus-Pytorch).

I recommend you measure the val-mIoU and test-mIoU without Resize_eval or use other DeepLabV3 implementations because our contribution in WSSS is the high-quality of pseudo-labels.

**Comment by Italy2006** [comment]
> For DeepLab-V2, we did not use Resize_eval.
> 
> The result of DeepLab-V3 is our additional experiment after publishing, so we skip the test-mIoU and report only the val-mIoU following the [DeepLabV3 implementation](https://github.com/VainF/DeepLabV3Plus-Pytorch).
> 
> I recommend you measure the val-mIoU and test-mIoU without Resize_eval or use other DeepLabV3 implementations because our contribution in WSSS is the high-quality of pseudo-labels.

Thank you for your reply. But I still have a question. In the article, the result of your Deeplabv2 on the validation set is 71.2%, but why is there only 70.4% in this library? In addition, is it convenient for you to upload the code for submitting the test set?
